### PR TITLE
[ADD][12.0] l10n_fi / Finnish Localization

### PR DESCRIPTION
**Pull request by mlaitinen**
**Description of the issue/feature this PR addresses:**
Finnish Localization

**Current behavior before PR:**
No localization for Finland

**Desired behavior after PR is merged:**
There will be a Finnish "Raportointikoodisto" chart of accounts available with:

- Descriptive names for taxes, accounts and fiscal positions
- Working tax and account replacement with fiscal positions in intra-community trade (including deducting and carrying out on the tax report, as per the law)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_fi/data/account_account_template_data.xml [hunk 1]
```diff
@@ -0,0 +1,5403 @@
...
+        <field name="chart_template_id" ref="chart_template_fi_rapko"/>
+        <field name="reconcile" eval="False"/>
+        <field name="name">Katettu j&#228;lleenhankintavarauksella</field>
+    </record>
+    <record model="account.account.template" id="a116204">
+        <field name="code">116204</field>
+        <field name="user_type_id" ref="account.data_account_type_current_assets"/>
+        <field name="chart_template_id" ref="chart_template_fi_rapko"/>
+        <field name="reconcile" eval="False"/>
+        <field name="name">
```
Comment (jco-odoo) [inline addons/l10n_fi/data/account_account_template_data.xml]: Do we need all 700+ accounts?  It seems they are generic enough, but always good if you could check if we can skip some.  Also, csv is better for loading in large quantities.

File: addons/l10n_fi/data/account_account_template_data.xml [hunk 2]
```diff
@@ -0,0 +1,5403 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+    <record model="account.account.template" id="a10002">
+        <field name="code">10002</field>
+        <field name="user_type_id" ref="account.data_account_type_current_assets"/>
+        <field name="chart_template_id" ref="chart_template_fi_rapko"/>
+        <field name="reconcile" eval="False"/>
```
[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/l10n_fi/data/account_account_template_data.xml]: but if we put it in csv it does not matter

File: addons/l10n_fi/data/account_account_tag_data.xml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    </record>
+    <record model="account.account.tag" id="tax_code_base_purchase_non_ic_tag">
+        <field name="applicability">taxes</field>
+        <field name="name">Ostot ei-yhteis&#246; 0 %</field>
+    </record>
+
+    <!-- Detailed Tax Codes -->
+    <record model="account.account.tag" id="tax_code_taxes_sale_rev_tag">
+        <field name="applicability">taxes</field>
+        <field name="name">301.10: K&#228;&#228;nnetty verovelvollisuus</field>
```
[the remark this answers is not in the document]
Comment (mlaitinen) [reply addons/l10n_fi/data/account_account_tag_data.xml]: I can't remember if there's been any good reason to do so. I will change it.

File: addons/l10n_fi/data/account_account_template_data.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mlaitinen) [reply addons/l10n_fi/data/account_account_template_data.xml]: I generated this file using an old XLSX to XML converter tool I made (https://github.com/mlaitinen/odoo_xlsx2xml) and it included even the False values.

I will remove them.

File: addons/l10n_fi/data/account_account_template_data.xml (hunk 1 above)
Comment (mlaitinen) [reply addons/l10n_fi/data/account_account_template_data.xml]: I will consult our accounting specialist if we could cut off some accounts or possibly add them as child-CoA's.

**Comment by hsh-odoo** [comment]
for the record, this PR is linked to the issue #18608

File: addons/l10n_fi/data/account_account_tag_data.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mlaitinen) [reply addons/l10n_fi/data/account_account_tag_data.xml]: I fixed these now in all of the data files

File: addons/l10n_fi/data/account_account_template_data.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mlaitinen) [reply addons/l10n_fi/data/account_account_template_data.xml]: I removed all `reconcile: False` fields.

File: addons/l10n_fi/data/account_account_template_data.xml (hunk 1 above)
Comment (mlaitinen) [reply addons/l10n_fi/data/account_account_template_data.xml]: There's nothing really that could be left out. It's a generic CoA and yes, there are quite a few accounts, but there's no compact version available.

**Comment by jco-odoo** [comment]
Is this a normal practice to have different accounts for additions, reductions (, depreciations)?  Can not these be put in the same account?   (Someone who wants different accounts can still add them afterwards)  Is it a legal requirement?  

In your fiscal positions, sometimes the tax is simply dropped.  Should not there be a vat 0%?  (I think for the purchase side (e.g. does the amount need to be reported?))

**Comment by mlaitinen** [comment]
@jco-odoo The multiple addition and reduction accounts don’t probably seem to make a lot of sense, but they do, when the account numbering and account groups are taken into consideration. The account groups are a useful part of this kit, but I didn’t include them here because I still want to fine-tune them with our finance people.

In those fiscal positions where the tax is dropped; yes, the amount won’t be reported so it’s not necessary to have it there. In the intra-community trade there’s a tax that has two sub-taxes; adding and deducting, which works nicely because there are two fields on the tax report that need to be filled.

**Comment by jco-odoo** [comment]
@mlaitinen Even with the account groups, I think it is better to at least give the full name of the account.  (so when you choose it, you know at least to which one the addition/reduction applies)  Debit/credit can not go on the same account?

**Comment by jco-odoo** [comment]
@mlaitinen Does anyone know what needs to be done for the VAT report?  I can not find it back easily.

**Comment by mlaitinen** [comment]
Closing this PR as the Association of Finnish Accounting Firms has abandoned the Raportointikoodisto CoA. We will not continue the module development.

**Pull request closed without merging**